internal/pkg/composable/controller.go

```diff
@@ -65,7 +65,7 @@ func New(log *logger.Logger, c *config.Config, managed bool) (Controller, error)
 	contextProviders := map[string]*contextProviderState{}
 	for name, builder := range Providers.contextProviders {
 		pCfg, ok := providersCfg.Providers[name]
-		if ok && !pCfg.Enabled() {
+		if (ok && !pCfg.Enabled()) || (!ok && providersCfg.ProvidersDefaultDisable) {
```

Comment:
In case a provider is configured and the `ProvidersDefaultDisable` is set, should we continue or not?

Reply:
So in that test-case, what is the expected output?
```go
		{
			cfg: map[string]interface{}{
				"agent.providers.initial_default": "true",
				"providers": map[string]interface{}{
					"env": map[string]interface{}{
						"enabled": "false",
					},
                                 },
			},
		},

```